Engine/DataFeeds/FileSystemDataFeed.cs

```diff
@@ -345,13 +345,17 @@ private ISubscriptionEnumeratorFactory GetEnumeratorFactory(SubscriptionRequest
             {
                 if (request.Universe is UserDefinedUniverse)
                 {
-                    // Trigger universe selection when security added manually after Initialize
+                    // Trigger universe selection when security added/removed after Initialize
                     var universe = (UserDefinedUniverse) request.Universe;
                     universe.CollectionChanged += (sender, args) =>
                     {
-                        if (args.NewItems == null || _frontierUtc == DateTime.MinValue) return;
+                        IList items =
+                            args.Action == NotifyCollectionChangedAction.Add ? items = args.NewItems :
+                            args.Action == NotifyCollectionChangedAction.Remove ? items = args.OldItems : null;
```

Comment:
Be careful with nested ternary operators... They can be easily misread. Code readability should always trump code brevity.